src/NServiceBus.Transport.RabbitMQ.TransportTests/When_restarting_endpoint.cs

```diff
@@ -0,0 +1,108 @@
+﻿namespace NServiceBus.TransportTests
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using NUnit.Framework;
+    using Transport;
+
+    public class When_restarting_endpoint : NServiceBusTransportTest
+    {
+        [TestCase(TransportTransactionMode.None)]
+        [TestCase(TransportTransactionMode.ReceiveOnly)]
+        [TestCase(TransportTransactionMode.SendsAtomicWithReceive)]
+        [TestCase(TransportTransactionMode.TransactionScope)]
+        public async Task Should_handle_multiple_stop_calls_to_started_receiver(TransportTransactionMode transactionMode)
+        {
+            await StartPump(
+                (context, token) => Task.CompletedTask,
+                (context, token) => Task.FromResult(ErrorHandleResult.Handled),
+                transactionMode);
+
+            await receiver.StopReceive();
+            await receiver.StopReceive();
+        }
+
+        [TestCase(TransportTransactionMode.None)]
+        [TestCase(TransportTransactionMode.ReceiveOnly)]
+        [TestCase(TransportTransactionMode.SendsAtomicWithReceive)]
+        [TestCase(TransportTransactionMode.TransactionScope)]
+        public async Task Should_allow_restarting_receivers(TransportTransactionMode transactionMode)
+        {
+            var messageReceived = CreateTaskCompletionSource();
+
+            await StartPump((context, token) =>
+            {
+                messageReceived.SetResult();
+                return Task.CompletedTask;
+            },
+                (context, token) => Task.FromResult(ErrorHandleResult.Handled), transactionMode);
+
+            await receiver.StopReceive();
+            await receiver.StartReceive();
+
+            await SendMessage(InputQueueName);
+            await messageReceived.Task;
```

Comment:
Is there something we could assert here? Not religious about it. Just a thought.

Reply:
hm not sure what else other than an invocation counter would make sense here?